Hardware/Tessel_2_Overview.md

```diff
@@ -0,0 +1,84 @@
+
+![](http://67.media.tumblr.com/83018b7f2595c1f4b3877f74eb86d134/tumblr_inline_nkz56hKzbm1s75tgz.png)
+
+These can be abstracted as follows:
+
+![](http://65.media.tumblr.com/559c4d45074644a5c335987bec7c698e/tumblr_inline_nkz56bL5SC1s75tgz.png)
+
+### Abstraction boundaries of Tessel 2
+
+The board employs a processor/coprocessor architecture. The Mediatek runs your user code, hosts USB devices, handles the network connections (be they wired, wireless, or cellular over USB), and communicates with the SAMD21.
+
+The SAM acts as a coprocessor and handles real-time, low-level IO through the module ports, USB comms through the Micro USB port, and programming the device as a whole.
+
+The two chips are connected by a SPI bridge that also includes the onboard flash (the [readme for Tessel 2’s firmware repo goes into more detail here](https://github.com/tessel/v2-firmware/blob/master/README.md)).
+
+The whole system is powered from the single Micro USB (device) port.
+
+## Power architecture of Tessel 2
+
+The below table explains how the major components of Tessel 2 are powered.
+
+Component          | Powered by       | Enabled or reset by
+-------------------|------------------|-------------------------------------------------
+5V                 | USB cable        | Presence of USB cable with upstream power
+3.3V Mediatek      | 5V               | GPIO on coprocessor, pull resistor to ENABLE
+Mediatek chip      | 3.3V Mediatek    | GPIO on coprocessor, pull resistor to !RESET
+Flash chip         | 3.3V Mediatek    | GPIO on both the SAM and the Mediatek
+1.8V               | 5V               | 3.3V Mediatek
+RAM for Mediatek   | 1.8V             | Mediatek (involuntary)
+3.3V coprocessor   | 5V               | 5V
+Coprocessor chip   | 3.3V Coprocessor | GPIO on Mediatek, pull resistor to !RESET
+3.3V module port A | 3.3V Coprocessor | GPIO on coprocessor, pull resistor to OFF
+3.3V module port B | 3.3V Coprocessor | GPIO on coprocessor, pull resistor to OFF
+USB hub IC         | 5V               | 3.3V Mediatek
+Power to USB ports | 5V               | Mediatek via USB hub IC, pull resistor to off
+
+What this means in practice is that the coprocessor (SAMD21) has control over both the "power plug" and the reset switch for the Mediatek, while the Mediatek only has control over the SAM's reset line. The rationale here is that the SAM uses a negligible amount of power, but it might, as you suggest, be beneficial to be able to shut down the Mediatek in order to save power.
+
+That said, there is no true "off switch" in the hardware that enables a permanent shutdown so long as external power is applied. Quiescent draw is currently estimated to be <10mW.
+
+With the default firmware in place, triggering a Mediatek reset from the SAM is...discouraged, but results in a clean reboot. I believe the SAM does currently save some state, but it can be reset by the Mediatek, so this isn't really a concern.
```

Comment:
It's a hard reset of the chip, not a proper clean shutdown. Of course, you do the same thing when you pull the USB cable.